File: RockitCompiler/benchmarks/bench_csv_pipeline.js
let seed = 42n;
let totalSum = 0n;
const n = 500000;
const cols = 100;
for (let row = 0; row < n; row++) {
    let col10 = 0n, col50 = 0n, col90 = 0n;
    for (let c = 0; c < cols; c++) {
        seed = (seed * 1103515245n + 12345n) % 2147483648n;
        const v = seed % 1000n;
        if (c === 10) col10 = v;
        if (c === 50) col50 = v;
        if (c === 90) col90 = v;
    }
    totalSum += col10 + col50 + col90;
}
console.log(totalSum.toString());

File: RockitCompiler/benchmarks/bench_csv_pipeline.cpp
#include <cstdio>
#include <cstdint>

int main() {
    int64_t n = 500000;
    int64_t cols = 100;
    int64_t seed = 42;
    int64_t totalSum = 0;
    for (int64_t row = 0; row < n; row++) {
        int64_t col10 = 0, col50 = 0, col90 = 0;
        for (int64_t c = 0; c < cols; c++) {
            seed = (seed * 1103515245 + 12345) % 2147483648LL;
            int64_t v = seed % 1000;
            if (c == 10) col10 = v;
            if (c == 50) col50 = v;
            if (c == 90) col90 = v;
        }
        totalSum += col10 + col50 + col90;
    }
    printf("%lld\n", totalSum);
    return 0;
}
